cdap-ui/app/cdap/components/DataPrep/WorkspaceTabs/index.js

```diff
@@ -98,129 +96,135 @@ export default class WorkspaceTabs extends Component {
         }
         return;
       }
+
+      this.setState({ deleteWorkspace: null });
+
       this.getWorkspaceList();
 
     }, (err) => {
       console.log("Error Deleting", err);
     });
   }
 
-  next() {
-    if (this.state.beginIndex + MAX_NUM_TABS >= this.state.workspaceList.length) { return; }
-
-    this.setState({
-      beginIndex: this.state.beginIndex + 1
-    });
+  toggleDeleteWorkspace(workspace) {
+    this.setState({deleteWorkspace: workspace});
   }
 
-  prev() {
-    if (this.state.beginIndex <= 0) { return; }
+  renderDropdown() {
+    if (this.state.workspaceList.length <= this.state.maxTabs) { return null; }
 
-    this.setState({
-      beginIndex: this.state.beginIndex - 1
-    });
-  }
+    let list = this.state.workspaceList.slice(this.state.maxTabs);
 
-  renderActiveWorkspace(workspace) {
-    return (
-      <div
-        key={workspace.id}
-        className="workspace-tab active"
-      >
-        <span title={workspace.name}>
-          {workspace.name}
-        </span>
-
-        <span
-          className="fa fa-fw delete-workspace"
-          onClick={this.deleteWorkspace.bind(this, workspace.id)}
-        >
-          <IconSVG
-            name="icon-close"
-          />
-        </span>
-      </div>
-    );
-  }
+    let tetherConfig = {
+      classes: {
+        element: 'workspace-list-popover'
+      }
+    };
 
-  renderInactiveWorkspace(workspace) {
     return (
-      <div
-        key={workspace.id}
-        className="workspace-tab"
-      >
-        <Link
-          to={`/ns/${this.namespace}/dataprep/${workspace.id}`}
-          title={workspace.name}
+      <div className="workspace-tab workspace-dropdown text-xs-center">
+        <UncontrolledPopover
+          tetherOption={tetherConfig}
         >
-          {workspace.name}
-        </Link>
-
-        <span
-          className="fa fa-fw delete-workspace"
-          onClick={this.deleteWorkspace.bind(this, workspace.id)}
-        >
-          <IconSVG
-            name="icon-close"
-          />
-        </span>
+          {
+            list.map((workspace) => {
+              return (
+                <div
+                  key={workspace.id}
+                  className="workspace-list-dropdown-item"
+                >
```

Comment:
Would be safer if add a `clearFix` and `float-xs-left` on the other element as we use `float-xs-right` on one of the elements. 

Seeing this in firefox
<img width="869" alt="screen shot 2017-05-24 at 6 43 04 pm" src="https://cloud.githubusercontent.com/assets/1452845/26432509/16daa3cc-40b1-11e7-9737-47d01cc59495.png">